lib/features/composer/presentation/composer_controller.dart

```diff
@@ -859,6 +866,94 @@ class ComposerController extends BaseController
     }
   }
 
+  Future<String> getTextOnlyContentInEditor() async {
+    try {
+      final htmlContent = await getContentInEditor();
+
+      String textContent = htmlContent.replaceAll(RegExp(r'<[^>]*>'), '');
+
+      textContent = textContent
+        .replaceAll('&nbsp;', ' ')
+        .replaceAll('&amp;', '&')
+        .replaceAll('&lt;', '<')
+        .replaceAll('&gt;', '>')
+        .replaceAll('&quot;', '"')
+        .replaceAll('&#39;', "'");
+
+      return textContent.trim();
+    } catch (e) {
+      logError('ComposerController::getTextOnlyContentInEditor:Exception = $e');
+      return '';
+    }
+  }
+
+  void insertTextInEditor(String text) {
+    final htmlContent = text.replaceAll('\n', '<br>');
+
+    if (PlatformInfo.isWeb) {
+      richTextWebController?.editorController.insertHtml(htmlContent);
+    } else {
+      richTextMobileTabletController?.htmlEditorApi?.insertHtml(htmlContent);
+    }
+  }
+
+  void showAIScribeMenuForFullText(BuildContext context) async {
+    final fullText = await getTextOnlyContentInEditor();
+
+    final RenderBox? renderBox = aiScribeButtonKey.currentContext?.findRenderObject() as RenderBox?;
+    Offset? buttonPosition;
+    if (renderBox != null) {
+      buttonPosition = renderBox.localToGlobal(Offset.zero);
+    }
+  
+    if (!context.mounted) return;
+
+    showAIScribeDialog(
+      context: context,
+      imagePaths: imagePaths,
+      content: fullText,
+      onInsertText: insertTextInEditor,
+      buttonPosition: buttonPosition,
+    );
+  }
```

Comment:
Should be written in a mixin or a widget so it can be reused for other views besides composer